include/screen.h

```diff
@@ -1,589 +1,661 @@
+//
+// This file is part of the NightDriver software project.
+//
+//    NightDriver is free software: you can redistribute it and/or modify
+//    it under the terms of the GNU General Public License as published by
+//    the Free Software Foundation, either version 3 of the License, or
+//    (at your option) any later version.
+//
+//    NightDriver is distributed in the hope that it will be useful,
+//    but WITHOUT ANY WARRANTY; without even the implied warranty of
+//    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//    GNU General Public License for more details.
+//
+//    You should have received a copy of the GNU General Public License
+//    along with Nightdriver.  It is normally found in copying.txt
+//    If not, see <https://www.gnu.org/licenses/>.
+//
+// Description:
+//
+//    Generalizes drawing to various different screens.  For example,
+//    drawing a line accepts a color in some cases, but not others, and
+//    it depends on what display you are compiling for.  This is a bit
+//    of an abstraction layer on those various devices.
+//
+// History:     Dec-10-2022         Davepl      Created
+//
+//---------------------------------------------------------------------------
+
+#pragma once
+
+#include <mutex>
+#include "gfxbase.h"
+#include <string>
+#include <vector>
+#include <memory>
+#if defined(TOGGLE_BUTTON_0) || defined(TOGGLE_BUTTON_1)
+    #include "Bounce2.h"
+#endif
+
+class Screen; // forward declaration for Page interface
+
+// Page
+//
+// Abstract base for renderable pages on the small status screen.
+// Implement Draw to render the page into the provided Screen. Optional
+// hooks handle button presses and whether the page should pause effect
+// rotation while visible.
+
+class Page
+{
+  public:
+    virtual ~Page() = default;
+
+    // Human-readable name (for diagnostics)
+    virtual std::string Name() const { return std::string("Page"); }
+
+    // Render this page. bRedraw indicates a full redraw is requested.
+    virtual void Draw(Screen& display, bool bRedraw) = 0;
+
+    // Optional button hook. Default behavior is provided in screen.cpp.
+    virtual void OnButtonPress(uint8_t buttonIndex);
+};
+
+class Screen : public GFXBase
+{
+  public:
+    static DRAM_ATTR std::mutex _screenMutex;
+
+    Screen(int w, int h) : GFXBase(w, h)
+    {
+    }
+
+    // Some devices, like the OLE, require that you send the whole buffer at once, but others do not.  The default implenetation is to do nothing.
+
+    virtual void StartFrame()
+    {
+    }
+
+    virtual void EndFrame()
+    {
+    }
+
+    // Display capabilities and theme colors
+    // Default: color display with a blue theme and white/yellow accents.
+    virtual bool IsMonochrome() const { return false; }
+    virtual uint16_t GetTextColor() const { return WHITE16; }
+    virtual uint16_t GetBkgndColor() const { return BLUE16; }
+    virtual uint16_t GetBorderColor() const { return YELLOW16; }

```

Comment:
Missed this one, before

```suggestion
        // Display capabilities and theme colors
        // Default: color display with a blue theme and white/yellow accents.
        virtual bool IsMonochrome() const 
        {
            return false;
        }

        virtual uint16_t GetTextColor() const
        {
            return WHITE16;
        }

        virtual uint16_t GetBkgndColor() const
        {
            return BLUE16;
        }

        virtual uint16_t GetBorderColor() const
        {
            return YELLOW16;
        }

```